custom/ilsgateway/tasks.py

```diff
@@ -150,13 +152,38 @@ def get_delivery_group_reports(domain, endpoint, facilities, checkpoint, date, l
         offset = 0
 
 
-# Temporary for staging
-@task(queue='background_queue', ignore_result=True)
-def ils_clear_stock_data_task():
-    StockTransaction.objects.filter(report__domain='ilsgateway-test-1').delete()
-    StockReport.objects.filter(domain='ilsgateway-test-1').delete()
-    products = Product.ids_by_domain('ilsgateway-test-1')
+@task(queue='background_queue')
+def ils_clear_stock_data_task(domain):
+    assert ILSGatewayConfig.for_domain(domain)
+    locations = SQLLocation.objects.filter(domain=domain)
+    SupplyPointStatus.objects.filter(supply_point__in=locations.values_list('location_id', flat=True)).delete()
+    DeliveryGroupReport.objects.filter(supply_point__in=locations.values_list('location_id', flat=True)).delete()
+    products = Product.ids_by_domain(domain)
     StockState.objects.filter(product_id__in=products).delete()
+    for supply_point in locations.exclude(supply_point_id__is_null=True).values_list('supply_point_id', flat=True):
+        StockTransaction.objects.filter(case_id=supply_point).delete()
+    StockReport.objects.filter(domain=domain).delete()
+    StockDataCheckpoint.objects.filter(domain=domain).delete()
+
+
+@task(queue='background_queue')
+def clear_report_data(domain):
+    locations_ids = SQLLocation.objects.filter(domain=domain).values_list('location_id', flat=True)
+    for location_id in locations_ids:
+        GroupSummary.objects.filter(org_summary__supply_point=location_id).delete()
```

Comment:
I think these can be slightly optimized by using `org_summary__supply_point__in=location_ids` and not bothering with the loop. does that not work?